Ui bandwidth cleanup

hunchback opened the pull request:
general code cleanup  of handling categorization according to active,warming, alert

safchain commented on config/config.go:
```diff
@@ -154,11 +154,11 @@ func init() {
 	cfg.SetDefault("ui.bandwidth_absolute_active", 1)
-	cfg.SetDefault("ui.bandwidth_absolute_alert", 1000)
 	cfg.SetDefault("ui.bandwidth_absolute_warning", 100)
+	cfg.SetDefault("ui.bandwidth_absolute_alert", 1000)
```
please do not change the order, it was in alphabetical order

hunchback replied:
ok

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1269,45 +1197,60 @@ TopologyGraphLayout.prototype = {
+      if (!link.source.visible || !link.target.visible)
+        continue;
+      if (link.metadata.RelationType != "layer2")
+	continue;
```
indentation issue

hunchback replied:
had a tab - replaced by space

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1269,45 +1197,60 @@ TopologyGraphLayout.prototype = {
         };
       } else {
-        delete this.linkLabels[link.id];
+	delete this.linkLabelData[link.id];
```
indentation

hunchback replied:
had a tab - replaced by space

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1269,45 +1197,60 @@ TopologyGraphLayout.prototype = {
       .classed ("link-label-alert", false);
     })
-    .remove();
+    exit.remove();
```
a ";" is missing the previous line or this "exit" has to be removed

hunchback replied:
fixed

safchain commented on statics/js/utils.js:
```diff
@@ -14,12 +14,15 @@ function debounce(func, wait, immediate) {
-    return (Math.floor(kbps / 1000)).toString() + " Mbps";
-  return kbps.toString() + " Kbps";
+function bandwidthToString(bps) {
+  const KBPS=1024, MBPS=1024*1024, GBPS = 1024*1024*1024;
```
space missing between "="

hunchback replied:
fixed

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1269,45 +1197,60 @@ TopologyGraphLayout.prototype = {
+    var self = this;
 
-    this.linkLabel = this.linkLabel.data(Object.values(this.linkLabels), function(d) { return d.id; });
+    this.updateLinkLabelData()
```
";" are missing

hunchback replied:
fixed

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1269,45 +1197,60 @@ TopologyGraphLayout.prototype = {
 
-    this.linkLabel = this.linkLabel.data(Object.values(this.linkLabels), function(d) { return d.id; });
+    this.updateLinkLabelData()
+    this.updateLinkLabel()
```
ditto

hunchback replied:
fixed

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1341,11 +1284,11 @@ TopologyGraphLayout.prototype = {
+    delete this.linkLabelData[link.id];
 
-    this.linkLabel = this.linkLabel.data(Object.values(this.linkLabels), function(d) { return d.id; });
+    this.updateLinkLabel()
```
":"

hunchback replied:
fixed

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1269,45 +1197,60 @@ TopologyGraphLayout.prototype = {
+
+      if (!link.source.visible || !link.target.visible)
+        continue;
+      if (link.metadata.RelationType != "layer2")
```
!==

hunchback commented:
@safchain - please note that I have addressed your comments and plus have added two related bug fixes:
- fixed case when TxBytes or RxBytes fields are missing
- partial fix in case clock skew (between agent and browser) - a "proper" fix would require using only the agent clock for calculations - to be addressed in a future PR

safchain commented on statics/js/components/graph-layout.js:
```diff
@@ -1154,7 +1154,12 @@ TopologyGraphLayout.prototype = {
+    // FIXME: calculation assumes browser clock (new Data()) is in
+    // sync with agent clock (new Date(metrics.Last)) which incorrect
+    // in the general case:
+    // (1) different timezones (no currently addressed)
```
for this last point we are using UTC time on the agent side, so I guess we should use UTC here too

safchain commented:
run cdd-overview
